examples/2d/elixir_advection_restart.jl

```diff
@@ -0,0 +1,32 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# create a restart file
+
+trixi_include(@__MODULE__, joinpath(@__DIR__, "elixir_advection_basic.jl"))
+
+
+###############################################################################
+# adapt the parameters that have changed compared to "elixir_advection_basic.jl"
+
+# Note: If you get a restart file from somewhere else, you need to provide
+# appropriate setups in the elixir loading a restart file
+
+restart_filename = joinpath("out", "restart_000040.h5")
+mesh = TreeMesh{2}(30_000) # n_cells_max
+load_mesh!(mesh, restart_filename)
```

Comment:
Why separate mesh generation and loading?

Reply:
That's also done in Taam: You need to pass the number of dimensions and `n_cells_max` there, so I combined them in the generation of the mesh.

Reply:
Makes sense. OTOH, is there a reason why we can't pass the capacity into `load_mesh`? Conceptually, I'd prefer it if the mesh type is implicitly set by the type of mesh that is referenced in the restart file. That way the user cannot specify a "wrong" mesh type. But if you think that's not necessary or too much effort to implement, just leave it as it is and close this thread :+1:

Reply:
Fixed in f992780.

Reply:
:+1: